File: misc/very-old-stuff/C/syracuse.c
#include <stdio.h>

int main(void)
{
    unsigned long iters;
    unsigned long value;
    unsigned long max_value;
    unsigned long score;
    unsigned long calc;
    unsigned char x;
    score = 0;
    iters = 0;
    for(value = 5; iters < 700; value++)
    {
        iters = 0;
        calc = value;
        //Calcul de la suite de syracuse.
        while(calc != 1)
        {
            if(calc & 0x01)
            {
                calc = (3 * calc + 1) / 2;
                iters++;
            }
            else
            {
                calc /= 2;
            }
            iters++;
        }
        if(iters > score)
        {
            score = iters;
            max_value = value;
            printf("%ld with %ld numbers.\n", value, iters);
        }
    }
}

File: misc/very-old-stuff/py/syracuse.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys

__author__ = "Florent Peterschmitt <fpeterscom@yahoo.fr>"

if __name__ == '__main__':
    value = 5
    score = 0
    max_value = 0
    iterations = 0
    while iterations < 700:
        b = value
        iterations = 0
        #Calcul de la suite jusqu'à arrivée à 1
        while b != 1:
            if b % 2 == 0:
                b = b / 2
            else:
                b = (3 * b + 1) / 2
                iterations += 1
            iterations += 1
        if iterations > score:
            score = iterations
            max_value = value
            print("{} with {} numbers".format(value, iterations))
        value += 1
    print("{} with {} numbers".format(max_value, score))
